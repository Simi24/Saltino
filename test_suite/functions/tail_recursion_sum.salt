// Test di sum con tail recursion
def main() {
    return sum_tail(100, 0)
}

def sum_tail(n, acc) {
    if (n <= 0) {
        return acc
    } else {
        return sum_tail(n - 1, acc + n)
    }
}
