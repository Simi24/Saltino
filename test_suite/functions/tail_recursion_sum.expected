Program result: 5050
